cmake/compiler/gcc/target.cmake

```diff
@@ -14,6 +14,29 @@ find_program(CMAKE_READELF    ${CROSS_COMPILE}readelf PATH ${TOOLCHAIN_HOME} NO_
 find_program(CMAKE_GDB        ${CROSS_COMPILE}gdb     PATH ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
 find_program(CMAKE_NM         ${CROSS_COMPILE}nm      PATH ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
 
+# x86_64 should pick up a proper cross compiler if one is provided,
+# but falling back to using the host toolchain is a very sane behavior
+# too.
+if(CONFIG_X86_64)
+  if(CMAKE_C_COMPILER STREQUAL CMAKE_C_COMPILER-NOTFOUND)
+    find_program(CMAKE_C_COMPILER   gcc    )
```

Comment:
BTW find_program() hits the cmake var CACHE first. So if CROSS_COMPILE is incomplete/partially broken then some of these find_program() will have no effect and the CMAKE_* end results will be partially from CROSS_COMPILE and partially from say /usr/bin.